### DISCUSSION TITLE
Where are images and vms stored?

### DISCUSSION THREAD

**Discussion by scastria** [rancher-sandbox/rancher-desktop#1551] [Q&A]
I would like to know where container images and VMs and other stuff are stored on my machine when running Rancher Desktop so that I can mark those folders as DO NOT BACKUP to save backup storage.

**Comment by mook-as** [comment]
This depends on the platform.

On Windows:
- `%USERPROFILE%\AppData\Local\rancher-desktop` contains the distribution data, container images, logs, etc.
- `%USERPROFILE%\AppData\Roaming\rancher-desktop` contains preferences.
- The application is installed to `%USERPROFILE%\AppData\Local\Programs\Rancher Desktop`

On macOS:
- `~/Library/Application Support/rancher-desktop` contains the bulk, including VMs (which include the container images).
- `~/Library/Preferences/rancher-desktop` contains preferences.
- `~/Library/Logs/rancher-desktop` contains… logs.
- `~/Library/Caches/rancher-desktop` contains caches (downloaded K3s distributions, etc.).
- We try to exclude the VMs from time machine backups; and caches should already be excluded.

On Linux:
- `~/.local/share/rancher-desktop` contains the VMs (including container images), logs, etc.
- `~/.config/rancher-desktop` contains preferences.
- `~/.cache/rancher-desktop` contains caches (downloaded K3s distributions, etc.).
- `~/.local/bin/` may contain symlinks to tools.
- We _try_ to follow [XDG specs](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) here; things can be overridden by environment variables.

Hope this helps!

**Reply by ivanmir** [reply]
Is there a way to make it store images and vms at different paths?
I couldn't find any settings to do so.

**Reply by jandubois** [reply]
There is no setting to do this. Images are always stored inside the VM, so by moving the VM storage, you move the images as well.

You can find a way to move the directory using a symlink on macOS in https://github.com/rancher-sandbox/rancher-desktop/issues/797#issuecomment-996510877. Linux should work the same way, just adjust the paths.

And I wrote an idea for Windows at https://github.com/rancher-sandbox/rancher-desktop/discussions/6296#discussioncomment-8089715, but got no feedback if it worked for the user or not (I haven't tested it myself).

Note that a factory-reset will undo this operation (and not delete the VM), so you would have to deal with that yourself if you ever do a reset.

**Reply by leonkosak** [reply]
@jandubois, thank you for explanations. 👍 
However, it would be highly beneficial to have a setting for storing VMs and settings.
For instance, some development environments are more strict and clear (delete) windows user profiles (and maybe also other locations) periodically.
And on Windows 11 (and later), there is a possibility to create Dev Drive (https://learn.microsoft.com/en-us/windows/dev-drive/).
Performance is also noticeably better because Windows Defender does not scan these drives and because of ReFS: https://blogs.windows.com/windowsdeveloper/2023/06/01/dev-drive-performance-security-and-control-for-developers/
Dev Drive is part of bigger story - Dev Home: https://learn.microsoft.com/en-us/windows/dev-home/

**Reply by jandubois** [reply]
@leonkosak I don't see how this is relevant to Rancher Desktop. It uses WSL2 and lets it pick the location for the distro. If you want WSL2 to put virtual volumes elsewhere, then it is up to you to configure it however you want; it is outside our control.

You may have noticed that on Windows you can't select the number of CPUs or amount of memory for the VM either. This is because WSL2 is running in a single Hyper-V instance, with the different distros just using different Linux namespaces. Since WSL2 settings are shared between distros, Rancher Desktop cannot control these settings without affecting the rest of your system.
